fitsio/fitsio_pywrap.c

```diff
@@ -4351,7 +4391,13 @@ PyFITSObject_read_header(struct PyFITSObject* self, PyObject* args) {
             add_string_to_dict(dict, "comment", scomment);
 
         } else {
-            was_converted = convert_keyword_to_allowed_ascii(keyname);
+
+            if (is_hierarch) {
```

Comment:
why are these treated differently?  Can you please add a comment?

Reply:
Yep. I will add a comment. Here is the logic

If a key is HIERARCH, then any ASCII character except `*?#` is allowed. If we don't do any conversion, then we can use the FITS key read functions, otherwise we have to write the value as a string.

If a key is not HIERARCH, then we have more cases. If a key has non-allowed chars for standard fits header keys, then we cannot use the FITS key read functions. So we track that. Also, if a key has non-ascii characters, then we need to convert it before we write it to the python dict. 

All of the if blocks here do that logic. I will add comments.